### ISSUE TITLE
Implementation of Side-Refinement

### ISSUE THREAD

**Issue by senliuy** [tianzhi0549/CTPN#58]
Hi, TianZhi,
I am trying to implement the training phase of CTPN with TensorFlow based on your code and faster-rcnn. However, I don't quietly understand the meaning of side-refinement or offset regression in your paper. 
The problems is I don't understand the meaning of x_{side} in your paper so I can't implement the training loss function.
Can you release the code of side-refinement or tell me how to calculate x_{side}? It will help me understanding the meaning of your excellent idea.
Many thanks!

**Comment by tianzhi0549** [comment]
Hi @senliuy, the side-refinement means that for every anchor, which is close to horizontal sides of word bounding boxes, need also regress the relative distance to the word bounding boxes. In other words, these anchors are required to output three values (y_c, h, x_side) instead of two values (y_c, h). Hope it is useful for you.

**Comment by senliuy** [comment]
Thanks, @tianzhi0549, you are very nice. I have finished the training code with your help. However, the convergence of my training seems not well. I took synthetic image as input. And the loss of classification is about 0.3 after 10k iterations. Is it correct? And may I ask the training loss of your experiment.

**Comment by senliuy** [comment]
Thanks, @tianzhi0549 . Now I have finished the code of CTPN.

**Comment by dajiangxiaoyan** [comment]
@senliuy  will you plan to release the code?

**Comment by senliuy** [comment]
@dajiangxiaoyan Sorry, the code is for company used currently.

**Comment by dajiangxiaoyan** [comment]
@senliuy  Thank you for reply. I also reimplement CTPN. But I have met two problem.
 1. when adding blstm layer after the last conv layer, the net is not converge. 
 2. how to split the label text bounding box. Just split it by 16 width. Fox example. the bbox is (x0,y0, x1, y1). Split it into (x0,y0,x0+16, y1)...?

**Comment by senliuy** [comment]
1. There are lots of reasons that lead to non-convergence. I'm sorry that I can't answer your question.
2. You are right, tianzhi gave the code of generating anchors. You can copy his code.

**Comment by dajiangxiaoyan** [comment]
@senliuy  Thank you very much. I will try.

**Comment by tonmoyborah** [comment]
I want to implement side refinement too. Any help on how I should proceed?

**Comment by hcnhatnam** [comment]
[Discuss about iplementation of Side-Refinement](https://github.com/tianzhi0549/CTPN/issues/97)
